org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/Scope.java

```diff
@@ -5144,13 +5156,17 @@ public int parameterCompatibilityLevel(MethodBinding method, TypeBinding[] argum
 					TypeBinding param = ((ArrayBinding) parameters[lastIndex]).elementsType();
 					for (int i = lastIndex; i < argLength; i++) {
 						TypeBinding arg = (tiebreakingVarargsMethods && (i == (argLength - 1))) ? ((ArrayBinding)arguments[i]).elementsType() : arguments[i];
-						if (TypeBinding.notEquals(param, arg) && parameterCompatibilityLevel(arg, param, env, tiebreakingVarargsMethods, method) == NOT_COMPATIBLE)
-							return NOT_COMPATIBLE;
+						if (TypeBinding.notEquals(param, arg)) {
+							level = parameterCompatibilityLevel(arg, param, env, tiebreakingVarargsMethods, method);
+							if (level == NOT_COMPATIBLE)
```

Comment:
What happens at the first call site of `org.eclipse.jdt.internal.compiler.lookup.Scope.parameterCompatibilityLevel(TypeBinding, TypeBinding, LookupEnvironment, boolean, MethodBinding)` in this method if it returns `COMPATIBLE_IGNORING_MISSING_TYPE` - AFAICT, if comparison of "standard arguments from 0 to lastIndex" succeeds we would return `COMPATIBLE` when the varargs type is actually missing ???

Reply:
Whenever we set `level` to `COMPATIBLE_IGNORING_MISSING_TYPE` there should not be a code path that will "improve" to `COMPATIBLE` or similar. Or do you see a path where `level==COMPATIBLE_IGNORING_MISSING_TYPE` can be overwritten with a non-negative value?

There's only one location where the result of `parameterCompatibilityLevel()` is not assigned ([line 5149](https://github.com/eclipse-jdt/eclipse.jdt.core/blob/285eee1ba99d6e7159ffbe3ea8f8f7eeb6eeae30/org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/Scope.java#L5149) above), but this can't be `COMPATIBLE_IGNORING_MISSING_TYPE` because with missing type we would not have entered the enclosing branch `if (level == NOT_COMPATIBLE)`.

With this, I believe we are clean. Feel free to demonstrate otherwise :)

Reply:
> parameterCompatibilityLevel

I checked again, you are right.  So this can be ignored.